sourmash_lib/kmer_min_hash.hh

```diff
@@ -156,29 +167,20 @@ public:
         return aa;
     }
 
-    std::string _checkdna(const char * s, bool force=false) const {
-        std::string seq = s;
-        const size_t seqsize = strlen(s);
+    bool _checkdna(std::string seq) const {
 
-        for (size_t i=0; i < seqsize; ++i) {
+        for (size_t i=0; i < seq.length(); ++i) {
```

Comment:
Would be nice to switch to `for (auto b : seq) {...}` as we touch this code. Up to you.